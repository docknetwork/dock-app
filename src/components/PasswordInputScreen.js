import React, {useState} from 'react';
import {
  Content,
  Footer,
  Header,
  Input,
  InputError,
  LoadingButton,
  NBox as Box,
  ScreenContainer,
  Typography,
} from '../design-system';
import {BackButton} from '../design-system/buttons';
import {translate} from '../locales';

export function PasswordInputScreen({
  form,
  onChange,
  submitDisabled,
  onSubmit,
}) {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('password_input.title')}
        </Typography>
        <Box mt={7}>
          <Input
            placeholder="Password"
            value={form.password}
            onChangeText={onChange('password')}
            autoCapitalize="none"
            secureTextEntry={true}
          />
        </Box>
        <InputError form={form} id="password" />
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <LoadingButton
          full
          testID="next-btn"
          isDisabled={submitDisabled}
          onPress={onSubmit}>
          {translate('password_input.submit')}
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}

export function PasswordInputContainer({onSubmit}) {
  const [form, setForm] = useState({
    password: '',
    _errors: {},
    _hasError: false,
  });

  const handleChange = key => evt => {
    setForm(v => ({
      ...v,
      [key]: evt,
    }));
  };

  const clearErrors = () =>
    setForm(f => ({
      _errors: {},
    }));

  const handleSubmit = async () => {
    try {
      clearErrors();
      await onSubmit(form);
    } catch (err) {
      console.error(err);
      setForm(f => ({
        ...f,
        _errors: {
          password: 'The password you entered is incorrect',
        },
      }));
    }
  };

  const submitDisabled = !form.password;

  return (
    <PasswordInputScreen
      form={form}
      submitDisabled={submitDisabled}
      onChange={handleChange}
      onSubmit={handleSubmit}
    />
  );
}
