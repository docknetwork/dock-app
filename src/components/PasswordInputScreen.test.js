import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {PasswordInputScreen} from './PasswordInputScreen';

const mockStore = configureMockStore();

describe('PasswordInputScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <PasswordInputScreen
        form={{
          password: 'test',
        }}
        onChange={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
