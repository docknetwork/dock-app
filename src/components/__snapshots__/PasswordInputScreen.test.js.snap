// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInputScreen should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#18181B",
      "flex": 1,
    }
  }
  testID="create-wallet-screen"
>
  <AppGlobalHeader />
  <Styled(View)>
    <BackButton />
  </Styled(View)>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Styled(Component)
      marginTop={52}
      variant="h1"
    >
      Enter password
    </Styled(Component)>
    <Memo(ForwardRef(Box))
      mt={7}
    >
      <Memo(ForwardRef(Input))
        autoCapitalize="none"
        placeholder="Password"
        secureTextEntry={true}
        value="test"
      />
    </Memo(ForwardRef(Box))>
    <InputError
      form={
        Object {
          "password": "test",
        }
      }
      id="password"
    />
  </Styled(ScrollView)>
  <Styled(Box)
    marginBottom={10}
    marginLeft={26}
    marginRight={26}
  >
    <LoadingButton
      full={true}
      testID="next-btn"
    >
      Next
    </LoadingButton>
  </Styled(Box)>
</RNCSafeAreaView>
`;
